blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -276,14 +276,33 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
 
 object BlazeClientBuilder {
 
+  sealed trait SSLContextOption
+  object SSLContextOption {
+    case object NoSSL extends SSLContextOption
+    case object TryDefaultSSLContext extends SSLContextOption
+    final case class Provided(sslContext: SSLContext) extends SSLContextOption
+
+    def toMaybeSSLContext(sco: SSLContextOption): Option[SSLContext] =
+      sco match {
+        case SSLContextOption.NoSSL => None
+        case SSLContextOption.TryDefaultSSLContext => tryDefaultSslContext
+        case SSLContextOption.Provided(context) => Some(context)
+      }
+  }
+
+  import SSLContextOption.{TryDefaultSSLContext, toMaybeSSLContext}
+
   /** Creates a BlazeClientBuilder
     *
     * @param executionContext the ExecutionContext for blaze's internal Futures
-    * @param sslContext Some `SSLContext.getDefault()`, or `None` on systems where the default is unavailable
+    * @param sslContextOption indicates how to resolve SSLContext. The sum type offers three options:
+    *                         NoSSL                = do not use SSL/HTTPS
+    *                         TryDefaultSSLContext = `SSLContext.getDefault()`, or `None` on systems where the default is unavailable
+    *                         Provided             = use the explicitly passed SSLContext
     */
   def apply[F[_]: ConcurrentEffect](
       executionContext: ExecutionContext,
-      sslContext: Option[SSLContext] = tryDefaultSslContext): BlazeClientBuilder[F] =
+      sslContextOption: SSLContextOption = TryDefaultSSLContext): BlazeClientBuilder[F] =
```

Comment:
I think this argument could and should go away.  We usually only have the things without sensible defaults in the builder `apply` methods.

Reply:
> I think this argument could and should go away.

Once I remove it, what would the default be, @rossabaker?

Reply:
Nevermind. I can see that `TryDefaultSSLContext` is the sensible default.

I've addressed your comment, @rossabaker, via https://github.com/http4s/http4s/pull/3422/commits/1122f6444ad1caa3b03b77587d1ddd51841cd36a.

Reply:
As a follow-up, do you think it's worthwhile to leave, but deprecate, the existing `apply(ExecutionContext, Option[SSLContext]`, @rossabaker?

Reply:
Yes.  You'll probably have to eliminate the default argument in the deprecated one, and `TryDefaultSSLContext` will just be the default on the builder itself.

Reply:
Deprecated - https://github.com/http4s/http4s/pull/3422/commits/4215c42dee5ce53bd4b1deb4aec9a7d52c081f48